{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2bcd6566",
   "metadata": {},
   "source": [
    "# Data and Study Design\n",
    "\n",
    "In our experiment we had two cohorts of mice. **Mice 2-5** received a fecal transplant from a **Healthy** human donor. **Mice 6-10** received am FMT from a human donor with **Ulcerative Colitis (UC)**. We refere to these mouse cohorts as the Healthy and UC cohorts.  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da30c9ba",
   "metadata": {},
   "source": [
    "An overview of the experiment is provided in the following figure\n",
    "![](figure/study_design.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "27ddfd51",
   "metadata": {},
   "source": [
    "In this notebook we will explore the two sources of data: qPCR for total bacterial load and reads from Amplicon sequencing. We will also go over the timing of fecal sample collection and perturbations as well as provide taxonomies for the associated amplicon sequences that were identified in the samples."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6acec74c",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "818b5e0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "lines_to_show=12"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b76a3d4d",
   "metadata": {},
   "source": [
    "\n",
    "## Metadata\n",
    "\n",
    "The following table is a metadata table.  For every sampleID it has the corresponding subject (mouse) and timepoint (in days) of fecal sample collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ff2766b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampleID</th>\n",
       "      <th>subject</th>\n",
       "      <th>time</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10-D0AM</td>\n",
       "      <td>10</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10-D10</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10-D11</td>\n",
       "      <td>10</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10-D14</td>\n",
       "      <td>10</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10-D16</td>\n",
       "      <td>10</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10-D18</td>\n",
       "      <td>10</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>10-D1AM</td>\n",
       "      <td>10</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10-D1PM</td>\n",
       "      <td>10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10-D21AM</td>\n",
       "      <td>10</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10-D21PM</td>\n",
       "      <td>10</td>\n",
       "      <td>21.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10-D22AM</td>\n",
       "      <td>10</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>10-D22PM</td>\n",
       "      <td>10</td>\n",
       "      <td>22.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sampleID  subject  time\n",
       "0    10-D0AM       10   0.0\n",
       "1     10-D10       10  10.0\n",
       "2     10-D11       10  11.0\n",
       "3     10-D14       10  14.0\n",
       "4     10-D16       10  16.0\n",
       "5     10-D18       10  18.0\n",
       "6    10-D1AM       10   1.0\n",
       "7    10-D1PM       10   1.5\n",
       "8   10-D21AM       10  21.0\n",
       "9   10-D21PM       10  21.5\n",
       "10  10-D22AM       10  22.0\n",
       "11  10-D22PM       10  22.5"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata = pd.read_csv('data/metadata.tsv', sep='\\t')\n",
    "metadata.head(lines_to_show)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "68506ec2",
   "metadata": {},
   "source": [
    "## qPCR\n",
    "\n",
    "\n",
    "The following table is a qPCR table. This has the replicate measurements (3) of the qPCR for every sample (already normalized by the mass of the fecal pellet) in the units of CFU/g.\n",
    "\n",
    "CFU/g: Colony Forming Units per gram of feces"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4e564875",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampleID</th>\n",
       "      <th>measurement1</th>\n",
       "      <th>measurement2</th>\n",
       "      <th>measurement3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10-D0AM</td>\n",
       "      <td>3.124446e+07</td>\n",
       "      <td>6.624073e+07</td>\n",
       "      <td>2.146892e+07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10-D10</td>\n",
       "      <td>1.249714e+11</td>\n",
       "      <td>1.476571e+11</td>\n",
       "      <td>5.866129e+10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10-D11</td>\n",
       "      <td>1.430439e+11</td>\n",
       "      <td>2.154554e+11</td>\n",
       "      <td>2.931997e+11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10-D14</td>\n",
       "      <td>7.274684e+10</td>\n",
       "      <td>1.950943e+11</td>\n",
       "      <td>6.949483e+10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10-D16</td>\n",
       "      <td>9.364484e+10</td>\n",
       "      <td>1.167788e+11</td>\n",
       "      <td>1.107184e+11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10-D18</td>\n",
       "      <td>5.760466e+10</td>\n",
       "      <td>7.984920e+10</td>\n",
       "      <td>1.032151e+11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>10-D1AM</td>\n",
       "      <td>8.787440e+09</td>\n",
       "      <td>1.503446e+10</td>\n",
       "      <td>1.081238e+10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10-D1PM</td>\n",
       "      <td>4.975814e+10</td>\n",
       "      <td>1.093237e+11</td>\n",
       "      <td>1.121214e+11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10-D21AM</td>\n",
       "      <td>1.305030e+11</td>\n",
       "      <td>2.303749e+11</td>\n",
       "      <td>1.680958e+11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10-D21PM</td>\n",
       "      <td>1.879675e+11</td>\n",
       "      <td>1.953849e+11</td>\n",
       "      <td>3.029993e+11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10-D22AM</td>\n",
       "      <td>9.745953e+10</td>\n",
       "      <td>8.361087e+10</td>\n",
       "      <td>1.168677e+11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>10-D22PM</td>\n",
       "      <td>4.495309e+10</td>\n",
       "      <td>8.937333e+10</td>\n",
       "      <td>8.074691e+10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sampleID  measurement1  measurement2  measurement3\n",
       "0    10-D0AM  3.124446e+07  6.624073e+07  2.146892e+07\n",
       "1     10-D10  1.249714e+11  1.476571e+11  5.866129e+10\n",
       "2     10-D11  1.430439e+11  2.154554e+11  2.931997e+11\n",
       "3     10-D14  7.274684e+10  1.950943e+11  6.949483e+10\n",
       "4     10-D16  9.364484e+10  1.167788e+11  1.107184e+11\n",
       "5     10-D18  5.760466e+10  7.984920e+10  1.032151e+11\n",
       "6    10-D1AM  8.787440e+09  1.503446e+10  1.081238e+10\n",
       "7    10-D1PM  4.975814e+10  1.093237e+11  1.121214e+11\n",
       "8   10-D21AM  1.305030e+11  2.303749e+11  1.680958e+11\n",
       "9   10-D21PM  1.879675e+11  1.953849e+11  3.029993e+11\n",
       "10  10-D22AM  9.745953e+10  8.361087e+10  1.168677e+11\n",
       "11  10-D22PM  4.495309e+10  8.937333e+10  8.074691e+10"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "qpcr = pd.read_csv('data/qpcr.tsv', sep='\\t')\n",
    "qpcr.head(lines_to_show)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36977f73",
   "metadata": {},
   "source": [
    "## Sequencing counts table - rows: ASVs, columns: samples\n",
    "\n",
    "The following table is a read counts table. This shows the number of reads associated with each ASV for each sample ID."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "5de185b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>10-D0AM</th>\n",
       "      <th>10-D10</th>\n",
       "      <th>10-D11</th>\n",
       "      <th>10-D14</th>\n",
       "      <th>10-D16</th>\n",
       "      <th>10-D18</th>\n",
       "      <th>10-D1AM</th>\n",
       "      <th>10-D1PM</th>\n",
       "      <th>10-D21AM</th>\n",
       "      <th>...</th>\n",
       "      <th>9-D60AM</th>\n",
       "      <th>9-D60PM</th>\n",
       "      <th>9-D61</th>\n",
       "      <th>9-D62</th>\n",
       "      <th>9-D63</th>\n",
       "      <th>9-D64AM</th>\n",
       "      <th>9-D64PM</th>\n",
       "      <th>9-D7</th>\n",
       "      <th>9-D8</th>\n",
       "      <th>9-D9</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASV_1</td>\n",
       "      <td>22</td>\n",
       "      <td>10503</td>\n",
       "      <td>21726</td>\n",
       "      <td>25990</td>\n",
       "      <td>21572</td>\n",
       "      <td>21352</td>\n",
       "      <td>17823</td>\n",
       "      <td>16916</td>\n",
       "      <td>26669</td>\n",
       "      <td>...</td>\n",
       "      <td>23613</td>\n",
       "      <td>20729</td>\n",
       "      <td>37601</td>\n",
       "      <td>38306</td>\n",
       "      <td>21944</td>\n",
       "      <td>12971</td>\n",
       "      <td>24560</td>\n",
       "      <td>12622</td>\n",
       "      <td>10323</td>\n",
       "      <td>10960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASV_2</td>\n",
       "      <td>21</td>\n",
       "      <td>9319</td>\n",
       "      <td>17515</td>\n",
       "      <td>20188</td>\n",
       "      <td>14972</td>\n",
       "      <td>14868</td>\n",
       "      <td>6627</td>\n",
       "      <td>7697</td>\n",
       "      <td>20212</td>\n",
       "      <td>...</td>\n",
       "      <td>10875</td>\n",
       "      <td>11709</td>\n",
       "      <td>6582</td>\n",
       "      <td>7369</td>\n",
       "      <td>4935</td>\n",
       "      <td>7152</td>\n",
       "      <td>15477</td>\n",
       "      <td>7814</td>\n",
       "      <td>9361</td>\n",
       "      <td>8009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV_3</td>\n",
       "      <td>3</td>\n",
       "      <td>9380</td>\n",
       "      <td>17748</td>\n",
       "      <td>20899</td>\n",
       "      <td>15247</td>\n",
       "      <td>17260</td>\n",
       "      <td>45</td>\n",
       "      <td>1078</td>\n",
       "      <td>21997</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>6</td>\n",
       "      <td>120</td>\n",
       "      <td>21</td>\n",
       "      <td>502</td>\n",
       "      <td>1034</td>\n",
       "      <td>5944</td>\n",
       "      <td>19830</td>\n",
       "      <td>15778</td>\n",
       "      <td>22068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASV_4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>24</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>19</td>\n",
       "      <td>25</td>\n",
       "      <td>23</td>\n",
       "      <td>24</td>\n",
       "      <td>20</td>\n",
       "      <td>45</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASV_5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ASV_6</td>\n",
       "      <td>0</td>\n",
       "      <td>3066</td>\n",
       "      <td>3391</td>\n",
       "      <td>3617</td>\n",
       "      <td>3919</td>\n",
       "      <td>3874</td>\n",
       "      <td>0</td>\n",
       "      <td>26</td>\n",
       "      <td>4377</td>\n",
       "      <td>...</td>\n",
       "      <td>3496</td>\n",
       "      <td>3423</td>\n",
       "      <td>1102</td>\n",
       "      <td>5511</td>\n",
       "      <td>5398</td>\n",
       "      <td>3471</td>\n",
       "      <td>9295</td>\n",
       "      <td>1722</td>\n",
       "      <td>2431</td>\n",
       "      <td>2276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ASV_7</td>\n",
       "      <td>0</td>\n",
       "      <td>1612</td>\n",
       "      <td>2871</td>\n",
       "      <td>3172</td>\n",
       "      <td>2444</td>\n",
       "      <td>2769</td>\n",
       "      <td>922</td>\n",
       "      <td>1222</td>\n",
       "      <td>3760</td>\n",
       "      <td>...</td>\n",
       "      <td>1903</td>\n",
       "      <td>2187</td>\n",
       "      <td>1153</td>\n",
       "      <td>1280</td>\n",
       "      <td>846</td>\n",
       "      <td>1357</td>\n",
       "      <td>2866</td>\n",
       "      <td>1314</td>\n",
       "      <td>1482</td>\n",
       "      <td>1336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ASV_8</td>\n",
       "      <td>14</td>\n",
       "      <td>404</td>\n",
       "      <td>842</td>\n",
       "      <td>1022</td>\n",
       "      <td>469</td>\n",
       "      <td>346</td>\n",
       "      <td>380</td>\n",
       "      <td>453</td>\n",
       "      <td>273</td>\n",
       "      <td>...</td>\n",
       "      <td>56</td>\n",
       "      <td>55</td>\n",
       "      <td>16</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>27</td>\n",
       "      <td>89</td>\n",
       "      <td>272</td>\n",
       "      <td>532</td>\n",
       "      <td>548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ASV_9</td>\n",
       "      <td>0</td>\n",
       "      <td>2125</td>\n",
       "      <td>3261</td>\n",
       "      <td>3806</td>\n",
       "      <td>2642</td>\n",
       "      <td>2598</td>\n",
       "      <td>10071</td>\n",
       "      <td>6941</td>\n",
       "      <td>1678</td>\n",
       "      <td>...</td>\n",
       "      <td>1821</td>\n",
       "      <td>2001</td>\n",
       "      <td>639</td>\n",
       "      <td>479</td>\n",
       "      <td>654</td>\n",
       "      <td>1666</td>\n",
       "      <td>3398</td>\n",
       "      <td>2390</td>\n",
       "      <td>3350</td>\n",
       "      <td>2630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ASV_10</td>\n",
       "      <td>3</td>\n",
       "      <td>852</td>\n",
       "      <td>1401</td>\n",
       "      <td>1768</td>\n",
       "      <td>1589</td>\n",
       "      <td>1835</td>\n",
       "      <td>2723</td>\n",
       "      <td>1757</td>\n",
       "      <td>2737</td>\n",
       "      <td>...</td>\n",
       "      <td>224</td>\n",
       "      <td>676</td>\n",
       "      <td>291</td>\n",
       "      <td>143</td>\n",
       "      <td>79</td>\n",
       "      <td>187</td>\n",
       "      <td>469</td>\n",
       "      <td>1214</td>\n",
       "      <td>1928</td>\n",
       "      <td>1682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ASV_11</td>\n",
       "      <td>0</td>\n",
       "      <td>1771</td>\n",
       "      <td>3587</td>\n",
       "      <td>4755</td>\n",
       "      <td>3913</td>\n",
       "      <td>3936</td>\n",
       "      <td>1398</td>\n",
       "      <td>2267</td>\n",
       "      <td>5223</td>\n",
       "      <td>...</td>\n",
       "      <td>127</td>\n",
       "      <td>215</td>\n",
       "      <td>40</td>\n",
       "      <td>86</td>\n",
       "      <td>87</td>\n",
       "      <td>209</td>\n",
       "      <td>798</td>\n",
       "      <td>2866</td>\n",
       "      <td>2479</td>\n",
       "      <td>3153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ASV_12</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>58</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>61</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12 rows Ã— 687 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  10-D0AM  10-D10  10-D11  10-D14  10-D16  10-D18  10-D1AM  \\\n",
       "0       ASV_1       22   10503   21726   25990   21572   21352    17823   \n",
       "1       ASV_2       21    9319   17515   20188   14972   14868     6627   \n",
       "2       ASV_3        3    9380   17748   20899   15247   17260       45   \n",
       "3       ASV_4        0       0       7      24       5      20        0   \n",
       "4       ASV_5        0       0       0       0       0       0        0   \n",
       "5       ASV_6        0    3066    3391    3617    3919    3874        0   \n",
       "6       ASV_7        0    1612    2871    3172    2444    2769      922   \n",
       "7       ASV_8       14     404     842    1022     469     346      380   \n",
       "8       ASV_9        0    2125    3261    3806    2642    2598    10071   \n",
       "9      ASV_10        3     852    1401    1768    1589    1835     2723   \n",
       "10     ASV_11        0    1771    3587    4755    3913    3936     1398   \n",
       "11     ASV_12        5       4       6       7       0       4       58   \n",
       "\n",
       "    10-D1PM  10-D21AM  ...  9-D60AM  9-D60PM  9-D61  9-D62  9-D63  9-D64AM  \\\n",
       "0     16916     26669  ...    23613    20729  37601  38306  21944    12971   \n",
       "1      7697     20212  ...    10875    11709   6582   7369   4935     7152   \n",
       "2      1078     21997  ...       21        6    120     21    502     1034   \n",
       "3         0         0  ...       22       19     25     23     24       20   \n",
       "4         0         0  ...       15       20      0      0      0        0   \n",
       "5        26      4377  ...     3496     3423   1102   5511   5398     3471   \n",
       "6      1222      3760  ...     1903     2187   1153   1280    846     1357   \n",
       "7       453       273  ...       56       55     16     17      9       27   \n",
       "8      6941      1678  ...     1821     2001    639    479    654     1666   \n",
       "9      1757      2737  ...      224      676    291    143     79      187   \n",
       "10     2267      5223  ...      127      215     40     86     87      209   \n",
       "11       41         0  ...        4        0      4      0      0        4   \n",
       "\n",
       "    9-D64PM   9-D7   9-D8   9-D9  \n",
       "0     24560  12622  10323  10960  \n",
       "1     15477   7814   9361   8009  \n",
       "2      5944  19830  15778  22068  \n",
       "3        45      8      0      0  \n",
       "4         0      0      0      0  \n",
       "5      9295   1722   2431   2276  \n",
       "6      2866   1314   1482   1336  \n",
       "7        89    272    532    548  \n",
       "8      3398   2390   3350   2630  \n",
       "9       469   1214   1928   1682  \n",
       "10      798   2866   2479   3153  \n",
       "11       15     61      0     11  \n",
       "\n",
       "[12 rows x 687 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts = pd.read_csv('data/counts.tsv', sep='\\t')\n",
    "counts.head(lines_to_show)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2d5f45c",
   "metadata": {},
   "source": [
    "Recall that read counts are not quantitative (but are instead relative to the read depth). The total read depth for each sample can be computed as"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "93cff577",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10-D0AM         84\n",
       "10-D10       46512\n",
       "10-D11       82066\n",
       "10-D14       96268\n",
       "10-D16       78009\n",
       "10-D18       82540\n",
       "10-D1AM      49805\n",
       "10-D1PM      48392\n",
       "10-D21AM    108082\n",
       "10-D21PM     70385\n",
       "10-D22AM     59366\n",
       "10-D22PM     73812\n",
       "dtype: object"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "read_depth=counts.sum()[1:-1] # first column was just the ASV number hence indexing from 1 to the end, removing the 0 index\n",
    "read_depth.head(12)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1e51be8b",
   "metadata": {},
   "source": [
    "The read depth for each sample changes, but the read depth does not correspond to changes in total bacterial load. That is why we also have a qPCR measurement for estimating total bacterial load in each sample."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e7fd38d",
   "metadata": {},
   "source": [
    "## ASV number, sequences and taxonomy\n",
    "\n",
    "The taxonomy table relates the ASV number to the actual amplicon sequence associated with the ASV number. In addition you have the taxonomic information down to the species level (if the classification is known). When multiple species are provided it means that those species all have the same ASV (and this specific region of the 16S rRNA gene is not enough to destinguish them) and if `NaN` then that ASV did not have an associated species in the training set or more than 5 species were returned."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d8bd23b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>sequence</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASV_1</td>\n",
       "      <td>TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Phocaeicola</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASV_2</td>\n",
       "      <td>TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>fragilis/ovatus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV_3</td>\n",
       "      <td>TACAGAGGTCTCAAGCGTTGTTCGGAATCACTGGGCGTAAAGCGTG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>Verrucomicrobiae</td>\n",
       "      <td>Verrucomicrobiales</td>\n",
       "      <td>Akkermansiaceae</td>\n",
       "      <td>Akkermansia</td>\n",
       "      <td>muciniphila</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASV_4</td>\n",
       "      <td>TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>cellulosilyticus/intestinalis/timonensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASV_5</td>\n",
       "      <td>TACAGAGGTCTCAAGCGTTGTTCGGAATCACTGGGCGTAAAGCGTG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>Verrucomicrobiae</td>\n",
       "      <td>Verrucomicrobiales</td>\n",
       "      <td>Akkermansiaceae</td>\n",
       "      <td>Akkermansia</td>\n",
       "      <td>muciniphila</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ASV_6</td>\n",
       "      <td>TACGTAGGGTGCGAGCGTTAATCGGAATTACTGGGCGTAAAGGGTG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Betaproteobacteria</td>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>Sutterellaceae</td>\n",
       "      <td>Parasutterella</td>\n",
       "      <td>excrementihominis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ASV_7</td>\n",
       "      <td>TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>fragilis/koreensis/kribbi/ovatus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ASV_8</td>\n",
       "      <td>TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>ovatus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ASV_9</td>\n",
       "      <td>TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>dorei/fragilis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ASV_10</td>\n",
       "      <td>TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>caccae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ASV_11</td>\n",
       "      <td>TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>salyersiae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ASV_12</td>\n",
       "      <td>TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>faecichinchillae/faecis/finegoldii/thetaiotaom...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      name                                           sequence   Kingdom  \\\n",
       "0    ASV_1  TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...  Bacteria   \n",
       "1    ASV_2  TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...  Bacteria   \n",
       "2    ASV_3  TACAGAGGTCTCAAGCGTTGTTCGGAATCACTGGGCGTAAAGCGTG...  Bacteria   \n",
       "3    ASV_4  TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...  Bacteria   \n",
       "4    ASV_5  TACAGAGGTCTCAAGCGTTGTTCGGAATCACTGGGCGTAAAGCGTG...  Bacteria   \n",
       "5    ASV_6  TACGTAGGGTGCGAGCGTTAATCGGAATTACTGGGCGTAAAGGGTG...  Bacteria   \n",
       "6    ASV_7  TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...  Bacteria   \n",
       "7    ASV_8  TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...  Bacteria   \n",
       "8    ASV_9  TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...  Bacteria   \n",
       "9   ASV_10  TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...  Bacteria   \n",
       "10  ASV_11  TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...  Bacteria   \n",
       "11  ASV_12  TACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAG...  Bacteria   \n",
       "\n",
       "             Phylum               Class               Order           Family  \\\n",
       "0     Bacteroidetes         Bacteroidia       Bacteroidales   Bacteroidaceae   \n",
       "1     Bacteroidetes         Bacteroidia       Bacteroidales   Bacteroidaceae   \n",
       "2   Verrucomicrobia    Verrucomicrobiae  Verrucomicrobiales  Akkermansiaceae   \n",
       "3     Bacteroidetes         Bacteroidia       Bacteroidales   Bacteroidaceae   \n",
       "4   Verrucomicrobia    Verrucomicrobiae  Verrucomicrobiales  Akkermansiaceae   \n",
       "5    Proteobacteria  Betaproteobacteria     Burkholderiales   Sutterellaceae   \n",
       "6     Bacteroidetes         Bacteroidia       Bacteroidales   Bacteroidaceae   \n",
       "7     Bacteroidetes         Bacteroidia       Bacteroidales   Bacteroidaceae   \n",
       "8     Bacteroidetes         Bacteroidia       Bacteroidales   Bacteroidaceae   \n",
       "9     Bacteroidetes         Bacteroidia       Bacteroidales   Bacteroidaceae   \n",
       "10    Bacteroidetes         Bacteroidia       Bacteroidales   Bacteroidaceae   \n",
       "11    Bacteroidetes         Bacteroidia       Bacteroidales   Bacteroidaceae   \n",
       "\n",
       "             Genus                                            Species  \n",
       "0      Phocaeicola                                                NaN  \n",
       "1      Bacteroides                                    fragilis/ovatus  \n",
       "2      Akkermansia                                        muciniphila  \n",
       "3      Bacteroides           cellulosilyticus/intestinalis/timonensis  \n",
       "4      Akkermansia                                        muciniphila  \n",
       "5   Parasutterella                                  excrementihominis  \n",
       "6      Bacteroides                   fragilis/koreensis/kribbi/ovatus  \n",
       "7      Bacteroides                                             ovatus  \n",
       "8      Bacteroides                                     dorei/fragilis  \n",
       "9      Bacteroides                                             caccae  \n",
       "10     Bacteroides                                         salyersiae  \n",
       "11     Bacteroides  faecichinchillae/faecis/finegoldii/thetaiotaom...  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "asv_and_taxonomy = pd.read_csv('data/asv_and_taxonomy.tsv', sep='\\t')\n",
    "asv_and_taxonomy.head(lines_to_show)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1c59dca0",
   "metadata": {},
   "source": [
    "## Perturbation type with start and stop time (days) for each subject"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19c44d67",
   "metadata": {},
   "source": [
    "Recall that there were three perturbations applied, High Fat Diet, Vancomycin, and Gentamicin. This table has the start and stop time (in days) for the perturbations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "cccab358",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>subject</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>21.5</td>\n",
       "      <td>28.5</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>21.5</td>\n",
       "      <td>28.5</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>21.5</td>\n",
       "      <td>28.5</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>21.5</td>\n",
       "      <td>28.5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>21.5</td>\n",
       "      <td>28.5</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>21.5</td>\n",
       "      <td>28.5</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>21.5</td>\n",
       "      <td>28.5</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>21.5</td>\n",
       "      <td>28.5</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>21.5</td>\n",
       "      <td>28.5</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Vancomycin</td>\n",
       "      <td>35.5</td>\n",
       "      <td>42.5</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Vancomycin</td>\n",
       "      <td>35.5</td>\n",
       "      <td>42.5</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Vancomycin</td>\n",
       "      <td>35.5</td>\n",
       "      <td>42.5</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             name  start   end  subject\n",
       "0   High Fat Diet   21.5  28.5        2\n",
       "1   High Fat Diet   21.5  28.5        3\n",
       "2   High Fat Diet   21.5  28.5        4\n",
       "3   High Fat Diet   21.5  28.5        5\n",
       "4   High Fat Diet   21.5  28.5        6\n",
       "5   High Fat Diet   21.5  28.5        7\n",
       "6   High Fat Diet   21.5  28.5        8\n",
       "7   High Fat Diet   21.5  28.5        9\n",
       "8   High Fat Diet   21.5  28.5       10\n",
       "9      Vancomycin   35.5  42.5        2\n",
       "10     Vancomycin   35.5  42.5        3\n",
       "11     Vancomycin   35.5  42.5        4"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "perturbations = pd.read_csv('data/perturbations.tsv', sep='\\t')\n",
    "perturbations.head(12)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e438cdd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "120d52f8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc20ab22",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9bb229b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e78ba707",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "py3",
   "language": "python",
   "name": "py3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
